File: content/realm/examples/java/local/app/src/androidTest/java/com/mongodb/realm/examples/java/RealmFileLocationTest.java
package com.mongodb.realm.examples.java;

import android.util.Log;

import com.mongodb.realm.examples.Expectation;
import com.mongodb.realm.examples.RealmTest;

import org.junit.Test;

import io.realm.Realm;
import io.realm.RealmConfiguration;

public class RealmFileLocationTest extends RealmTest {
    @Test
    public void testGetRealmFileLocation() {
        Expectation expectation = new Expectation();
        activity.runOnUiThread(() -> {
            RealmConfiguration config = new RealmConfiguration.Builder()
                    .allowQueriesOnUiThread(true)
                    .allowWritesOnUiThread(true)
                    .build();

            // :snippet-start: get-realm-file-location
            Realm realm = Realm.getInstance(config);
            Log.v("EXAMPLE", "Realm file path: " + realm.getPath());
            // :snippet-end:
            realm.close();
            expectation.fulfill();
        });
        expectation.await();
    }
}

File: content/realm/examples/java/local/app/src/androidTest/java/com/mongodb/realm/examples/kotlin/RealmFileLocationTest.kt
package com.mongodb.realm.examples.kotlin

import android.util.Log
import com.mongodb.realm.examples.Expectation
import com.mongodb.realm.examples.RealmTest
import io.realm.Realm
import io.realm.RealmConfiguration
import org.junit.Test

class RealmFileLocationTest : RealmTest() {
    @Test
    fun testGetRealmFileLocation() {
        val expectation = Expectation()
        activity!!.runOnUiThread {
            val config = RealmConfiguration.Builder()
                .allowQueriesOnUiThread(true)
                .allowWritesOnUiThread(true)
                .build()

            // :snippet-start: get-realm-file-location
            val realm = Realm.getInstance(config)
            Log.v("EXAMPLE", "Realm file path: ${realm.path}")
            // :snippet-end:
            realm.close()
            expectation.fulfill()
        }
        expectation.await()
    }
}